pkg/command/context.go

```diff
@@ -622,65 +624,107 @@ func globalLogin(c *configtypes.Context) (err error) {
 	return nil
 }
 
-func globalTanzuLogin(c *configtypes.Context) error {
-	var claims *csp.Claims
-	var err error
-	apiTokenValue, apiTokenExists := os.LookupEnv(config.EnvAPITokenKey)
-	// Use API Token login flow if TANZU_API_TOKEN environment variable is set, else fall back to default interactive login flow
-	if apiTokenExists {
-		log.Info("API token env var is set")
-		claims, err = doCSPAPITokenAuthAndUpdateContext(c, apiTokenValue)
-	} else {
-		claims, err = doCSPInteractiveLoginAndUpdateContext(c)
-	}
+func globalTanzuLogin(c *configtypes.Context, generateContextNameFunc func(orgName, endpoint string, isStaging bool) string) error {
+	claims, err := doCSPAuthentication(c)
 	if err != nil {
 		return err
 	}
-	c.AdditionalMetadata[config.OrgIDKey] = claims.OrgID
 
-	kubeCfg, kubeCtx, serverEndpoint, err := tanzuauth.GetTanzuKubeconfig(c, endpoint, claims.OrgID, endpointCACertPath, skipTLSVerify)
+	orgName, err := getCSPOrganizationName(c, claims)
 	if err != nil {
 		return err
 	}
+	// update the context name using the context name generator
+	if generateContextNameFunc != nil {
+		c.Name = generateContextNameFunc(orgName, endpoint, staging)
+	}
 
-	c.ClusterOpts.Path = kubeCfg
-	c.ClusterOpts.Context = kubeCtx
-	c.ClusterOpts.Endpoint = serverEndpoint
+	// update the context metadata
+	if err := updateTanzuContextMetadata(c, claims.OrgID, orgName); err != nil {
+		return err
+	}
 
-	err = config.AddContext(c, true)
-	if err != nil {
+	// Fetch the tanzu kubeconfig and update context
+	if err := updateContextWithTanzuKubeconfig(c, endpoint, claims.OrgID, endpointCACertPath, skipTLSVerify); err != nil {
+		return err
+	}
+
+	// Add the context to configuration
+	if err := config.AddContext(c, true); err != nil {
 		return err
 	}
+
 	// update the current context in the kubeconfig file after creating the context
-	err = syncCurrentKubeContext(c)
-	if err != nil {
+	if err := syncCurrentKubeContext(c); err != nil {
 		return errors.Wrap(err, "unable to update current kube context")
 	}
 
 	// format
 	fmt.Println()
```

Comment:
Nit: this printout goes to stdout while the `log.Successf` goes to stderr.  How about using `fmt.Fprintf(os.Stderr)`

Reply:
updated. Thanks